plugins/formal-verification/testData/diagnostics/bad_contracts/list.fir.diag.txt

```diff
@@ -0,0 +1,123 @@
+/list.kt:(91,105): info: Generated Viper text for empty_list_get:
+method global$fun_empty_list_get$fun_take$$return$T_Unit()
+  returns (ret: dom$Unit)
+{
+  var anonymous$0: Ref
+  var local0$myList: Ref
+  var anonymous$1: Int
+  var local0$s: Int
+  anonymous$0 := pkg$kotlin$collections$global$fun_emptyList$fun_take$$return$T_class_pkg$kotlin$collections$global$class_List()
+  local0$myList := anonymous$0
+  anonymous$1 := pkg$kotlin$collections$class_scope_pkg$kotlin$collections$global$class_List$fun_get$fun_take$T_class_pkg$kotlin$collections$global$class_List$T_Int$return$T_Int(local0$myList,
+    0)
+  local0$s := anonymous$1
+  label label$ret$0
+}
+
+method pkg$kotlin$collections$global$fun_emptyList$fun_take$$return$T_class_pkg$kotlin$collections$global$class_List()
+  returns (ret: Ref)
+  ensures dom$Type$isSubtype((dom$TypeOf$typeOf(ret): dom$Type), dom$Type$pkg$kotlin$collections$global$class_List())
+  ensures acc(ret.special$size, write)
+  ensures ret.special$size >= 0
+  ensures ret.special$size == 0
```

Comment:
Where does this repetition come from?  Can it be avoided?

Reply:
`size >= 0` is a type invariant for `Collection` objects and is added for every argument/receiver/return of type `Collection`, on the other hand, `size == 0` is a specific postcondition of the function `emptyList()`. I don't think we can avoid it in an easy way and is not a big problem having this repetition i think